prov/gni/src/gnix_tags.c

```diff
@@ -254,6 +294,149 @@ static inline struct gnix_tag_list_element *__tag_list_find_element(
 }
 
 /**
+ * @brief peeks into a tag list to find the first match using given parameters
+ *
+ * @param ts           pointer to gnix_tag_storage_object
+ * @param tag          tag to find
+ * @param ignore       bits to ignore in tags
+ * @param list         dlist to search
+ * @param flags        fi_tagged flags
+ * @param context      fi_context associated with tag
+ * @param addr         gnix_address to find
+ * @param addr_ignore  bits to ignore in address
+ * @return NULL, if no match is found,
+ *         a non-NULL value, if a match is found
+ */
+static inline struct gnix_tag_list_element *__tag_hlist_find_first_match(
+		struct gnix_tag_storage *ts,
+		uint64_t tag,
+		uint64_t ignore,
+		struct dlist_entry *list,
+		uint64_t flags,
+		void *context,
+		struct gnix_address *addr,
+		uint64_t oldest_seq,
+		uint64_t oldest_gen,
+		int (*match_func)(struct dlist_entry *entry, const void *arg))
+{
+	struct dlist_entry *current;
+	struct gnix_tag_list_element *tle;
+	struct gnix_tag_search_element s_elem = {
+			.tag = tag,
+			.ignore = ignore,
+			.flags = flags,
+			.context = context,
+			.use_src_addr_matching = ts->attr.use_src_addr_matching,
+			.addr = addr,
+	};
+
+	/* search the list for a matching element. stop at the first match */
+	dlist_foreach(list, current) {
+		tle = container_of(current, struct gnix_tag_list_element, free);
+		GNIX_INFO(FI_LOG_EP_CTRL, "found entry, "
+				"oldest=%llx:%llx current=%llx:%llx\n",
+				oldest_seq, oldest_gen, tle->seq, tle->gen);
```

Comment:
This info log seems a bit excessive (I use FI_LOG_LEVEL info regularly).  Seems more useful if it only printed out something for the following 2 conditionals as well as right before the `return NULL` below.